### DISCUSSION TITLE
Best-practice recommendations on authenticating to Gruntwork's codebase in my CI/CD pipelines

### DISCUSSION THREAD

**Discussion by iantybo** [gruntwork-io/knowledge-base#632] [Help] [answered]
Since there are many ways to do this, what is the best way to authenticate to the Gruntwork modules on GitHub.com in my CI/CD pipelines? Also, how does this impact my authentication to other module sources, such as my internal VCS? 

---

<ins datetime="2023-01-12T17:14:33Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109793">Tracked in ticket #109793</a></p>
</ins>

**Comment by zackproser** [comment, marked as answer]
[This Knowledge Base post](https://github.com/gruntwork-io/knowledge-base/discussions/650) discusses how ECS Deploy Runner and Gruntwork Pipelines use your GitHub Personal Access Token (PAT) securely, by storing it in AWS Secrets Manager and only fetching it into your running ECS container on a just-in-time basis, so your token only exists ephemerally in volatile memory within your running task. This is the default pattern that Gruntwork prefers to use when authenticating to your GitHub resources within your CI/CD pipelines.
